sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventFacadeEjb.java

```diff
@@ -1021,26 +1049,51 @@ public List<String> getArchivedUuidsSince(Date since) {
 
 	@Override
 	@RightsAllowed(UserRight._EVENT_ARCHIVE)
-	public void archive(String eventUuid, Date endOfProcessingDate) {
-		super.archive(eventUuid, endOfProcessingDate);
+	public ProcessedEntity archive(String eventUuid, Date endOfProcessingDate) {
+		ProcessedEntity processedEntity = super.archive(eventUuid, endOfProcessingDate);
 		List<String> eventParticipantList = eventParticipantService.getAllUuidsByEventUuids(Collections.singletonList(eventUuid));
 		eventParticipantService.archive(eventParticipantList);
+
+		if (processedEntity.getProcessedEntityStatus().equals(ProcessedEntityStatus.EXTERNAL_SURVEILLANCE_FAILURE)) {
+			throw new ExternalSurveillanceToolRuntimeException(
```

Comment:
why to throw exception? Handling the processed entity status results in cleaner code than a try/catch